libmamba/tests/src/util/test_string.cpp

```diff
@@ -72,6 +72,34 @@ namespace mamba::util
             CHECK(contains("", ""));  // same as Python ``"" in ""``
         }
 
+        TEST_CASE("remove_prefix")
+        {
+            CHECK_EQ(remove_prefix("", ""), "");
+            CHECK_EQ(remove_prefix("hello", ""), "hello");
+            CHECK_EQ(remove_prefix("hello", "hello"), "");
+            CHECK_EQ(remove_prefix("", "hello"), "");
+            CHECK_EQ(remove_prefix("https://localhost", "https://"), "localhost");
+            CHECK_EQ(remove_prefix("https://localhost", "http://"), "https://localhost");
+            CHECK_EQ(remove_prefix("aabb", "a"), "abb");
+            CHECK_EQ(remove_prefix("", 'a'), "");
+            CHECK_EQ(remove_prefix("a", 'a'), "");
+            CHECK_EQ(remove_prefix("aaa", 'a'), "aa");
```

Comment:
Does trying to remove a suffix with `remove_prefix` and vice versa work?
i.e: `CHECK_EQ(remove_prefix("aabb", "b"), "aabb");`